codegen/main.rs

```diff
@@ -12,7 +12,7 @@ fn main() {
     Thread::spawn(move || {
         let output_dir = Path::new(os::getenv("OUT_DIR").unwrap());
         read_method::generate(output_dir).unwrap();
-    }).detach();
+    });
```

Comment:
?

I think what is actually desired here is to `.join()` the guard at the end of the function. Or just ditch the threading, it's probably a silly idea anyway.